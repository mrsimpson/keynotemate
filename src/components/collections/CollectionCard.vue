<template>
  <div class="collection-card">
    <div class="card-header">
      <h2>{{ collection.metadata.name }}</h2>
      <p class="description">{{ collection.metadata.description }}</p>
    </div>

    <div class="song-list">
      <div
        v-for="song in collection.songs"
        :key="song.id"
        class="song-item"
        :class="{ imported: isImported(song.id) }"
      >
        <div class="song-info">
          <span class="song-name">{{ song.name }}</span>
          <div v-if="song.composer || song.arranger" class="song-details">
            <span v-if="song.composer" class="composer">{{
              song.composer
            }}</span>
            <span v-if="song.arranger" class="arranger"
              >arr. {{ song.arranger }}</span
            >
          </div>
        </div>
        <button
          class="btn btn-primary btn-sm"
          :disabled="isImported(song.id)"
          @click="$emit('import', collection.metadata.id, song)"
        >
          {{
            isImported(song.id)
              ? t('collections.added')
              : t('collections.addToLibrary')
          }}
        </button>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { useI18n } from 'vue-i18n';
import type { CollectionSong, MusicCollection } from '../../types/collection';

const { t } = useI18n();

const props = defineProps<{
  collection: MusicCollection;
  importedSongIds: Set<string>;
}>();

defineEmits<{
  (e: 'import', collectionId: string, song: CollectionSong): void;
}>();

const isImported = (songId: string) => props.importedSongIds.has(songId);
</script>

<style scoped>
.collection-card {
  background: white;
  border-radius: var(--radius-md);
  padding: var(--spacing-lg);
  box-shadow: var(--shadow-sm);
}

.card-header {
  border-bottom: 1px solid #eee;
  padding-bottom: var(--spacing-md);
}

h2 {
  margin: 0;
  color: var(--text);
}

.description {
  color: #666;
  margin-top: var(--spacing-sm);
  font-size: 0.9rem;
}

.song-list {
  margin-top: var(--spacing-lg);
}

.song-item {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: var(--spacing-md);
  border-radius: var(--radius-sm);
  background: var(--light);
  margin-bottom: var(--spacing-sm);
  transition: background var(--transition-normal);
}

.song-item:hover {
  background: #f0f0f0;
}

.song-item.imported {
  background: #f0f9f4;
}

.song-info {
  display: flex;
  flex-direction: column;
  gap: 2px;
}

.song-name {
  font-weight: 500;
}

.song-details {
  font-size: 0.8rem;
  color: #666;
  display: flex;
  gap: var(--spacing-sm);
}

.btn-sm {
  padding: var(--spacing-xs) var(--spacing-sm);
  font-size: 0.9rem;
  min-width: 120px;
}

.btn:disabled {
  background: #90cba6;
  cursor: default;
}

@media (max-width: 600px) {
  .song-item {
    flex-direction: column;
    align-items: stretch;
    gap: var(--spacing-sm);
  }

  .btn-sm {
    width: 100%;
  }
}
</style>
