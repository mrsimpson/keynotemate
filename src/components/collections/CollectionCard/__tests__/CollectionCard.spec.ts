import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { createI18n } from 'vue-i18n'
import CollectionCard from '@/components/collections/CollectionCard.vue'

describe('CollectionCard Component', () => {
  const i18n = createI18n({
    legacy: false,
    locale: 'en',
    messages: {
      en: {
        collections: {
          addToLibrary: 'Add to Library',
          added: 'Added',
        },
      },
    },
  })

  const mockCollection = {
    metadata: {
      id: 'test-collection',
      name: 'Test Collection',
      description: 'A test collection description',
    },
    songs: [
      {
        id: 'song1',
        name: 'Song 1',
        composer: 'Composer 1',
        arranger: 'Arranger 1',
        filename: 'song1.xml',
      },
      {
        id: 'song2',
        name: 'Song 2',
        filename: 'song2.xml',
      },
    ],
  }

  it('renders collection card snapshot', () => {
    const wrapper = mount(CollectionCard, {
      props: {
        collection: mockCollection,
        importedSongIds: new Set(),
      },
      global: {
        plugins: [i18n],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders collection card with imported songs snapshot', () => {
    const wrapper = mount(CollectionCard, {
      props: {
        collection: mockCollection,
        importedSongIds: new Set(['song1']),
      },
      global: {
        plugins: [i18n],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
