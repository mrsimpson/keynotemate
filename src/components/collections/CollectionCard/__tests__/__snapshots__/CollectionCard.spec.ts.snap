// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CollectionCard Component > renders collection card snapshot 1`] = `
"<div data-v-a67a9f3f="" class="collection-card">
  <div data-v-a67a9f3f="" class="card-header">
    <h2 data-v-a67a9f3f="">Test Collection</h2>
    <p data-v-a67a9f3f="" class="description">A test collection description</p>
  </div>
  <div data-v-a67a9f3f="" class="song-list">
    <div data-v-a67a9f3f="" class="song-item">
      <div data-v-a67a9f3f="" class="song-info"><span data-v-a67a9f3f="" class="song-name">Song 1</span>
        <div data-v-a67a9f3f="" class="song-details"><span data-v-a67a9f3f="" class="composer">Composer 1</span><span data-v-a67a9f3f="" class="arranger">arr. Arranger 1</span></div>
      </div><button data-v-a67a9f3f="" class="btn btn-primary btn-sm">Add to Library</button>
    </div>
    <div data-v-a67a9f3f="" class="song-item">
      <div data-v-a67a9f3f="" class="song-info"><span data-v-a67a9f3f="" class="song-name">Song 2</span>
        <!--v-if-->
      </div><button data-v-a67a9f3f="" class="btn btn-primary btn-sm">Add to Library</button>
    </div>
  </div>
</div>"
`;

exports[`CollectionCard Component > renders collection card with imported songs snapshot 1`] = `
"<div data-v-a67a9f3f="" class="collection-card">
  <div data-v-a67a9f3f="" class="card-header">
    <h2 data-v-a67a9f3f="">Test Collection</h2>
    <p data-v-a67a9f3f="" class="description">A test collection description</p>
  </div>
  <div data-v-a67a9f3f="" class="song-list">
    <div data-v-a67a9f3f="" class="song-item imported">
      <div data-v-a67a9f3f="" class="song-info"><span data-v-a67a9f3f="" class="song-name">Song 1</span>
        <div data-v-a67a9f3f="" class="song-details"><span data-v-a67a9f3f="" class="composer">Composer 1</span><span data-v-a67a9f3f="" class="arranger">arr. Arranger 1</span></div>
      </div><button data-v-a67a9f3f="" class="btn btn-primary btn-sm" disabled="">Added</button>
    </div>
    <div data-v-a67a9f3f="" class="song-item">
      <div data-v-a67a9f3f="" class="song-info"><span data-v-a67a9f3f="" class="song-name">Song 2</span>
        <!--v-if-->
      </div><button data-v-a67a9f3f="" class="btn btn-primary btn-sm">Add to Library</button>
    </div>
  </div>
</div>"
`;
